### ISSUE TITLE
Move current GoCD templates into Yaml. Or find a way to reference global Yaml templates

### ISSUE THREAD

**Issue by sebawagner** [tomzo/gocd-yaml-config-plugin#132]
Hi,

I saw various discussion of how to do templating using GoCD and Yaml. 

However they seem to be different to what GoCD does. Basically I would just like to move existing templates from the "template" section into Yaml. For the same reason why doing pipelines in Yaml (version control, performs better then having infinite growing config.xml, easy of use).

The other issues raised against the gocd-yaml-config-plugin project seem to come up with new alternative ways of templating. Which may be a valid idea. But I think its yet another abstraction on top of an already abstract concept. 

Basically I think the easiest would be to just re-create the existing functionality that GoCD has. 

How hard would it be to inject Yaml templates into the current template section of the config.xml ? Or is that even somehow possible already ?

**Comment by tomzo** [comment]
Hi @sebawagner 

> Basically I think the easiest would be to just re-create the existing functionality that GoCD has.

Unfortunately, this is very hard due to the way xml config is implemented on the GoCD side. It is probably easier to built an independent template mechanism in plugins rather than changing the core config implementation. Also, making plugins capable of expanding templates allows to roll out more implementations.

> How hard would it be to inject Yaml templates into the current template section of the config.xml ? Or is that even somehow possible already ?

Why would would you want to do that? I thought that you want to move the templates from the central xml config to external repositories. I'm also not sure what you mean by 'Yaml templates' in this case.

You might want to see the discussion on https://github.com/gocd/gocd/issues/5675

**Comment by sebawagner** [comment]
@tomzo yeah gocd/gocd#5675 is basically what I meant

There is a comment in there that pretty much sums it up:
> The only piece we are missing from the current template feature is the ability to have them in source control rather than in the main GoCD config XML.
See: https://github.com/gocd/gocd/issues/5675#issuecomment-461022820

I agree, best would be having gocd/gocd#5675

**Comment by lobsterdore** [comment]
@sebawagner you can use Gomatic to configure Templates from code - https://github.com/gocd-contrib/gomatic, that's what we have ended up doing due to the lack of Template support via YAML. We configure a lot of our pipelines using Gomatic, allowing us to offer a devs a very simple interface for creating new service pipelines.

**Comment by kniec** [comment]
@lobsterdore do you happen to have an example of how you use it?

**Comment by tomzo** [comment]
I'm closing this as a duplicate of #2
